packages/deployer/src/deployment.js

```diff
@@ -39,7 +39,6 @@ class Deployment {
 
     const overwrite = isObject && isDeployed && lastArg.overwrite === false;
 
-    isObject && delete lastArg.overwrite;
```

Comment:
What an odd side effect! 😞

Reply:
Thanks for your response. Reproducing steps added to the description.

Reply:
Thank you! I was able to reproduce the error and verify the fix! Previously, I had a `value` property in the `txParams` object.